libraries/SITL/SIM_Frame.cpp

```diff
@@ -136,7 +139,9 @@ static Motor hexa_cw_x_motors[] =
     Motor(AP_MOTORS_MOT_5,  -90, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 5),
     Motor(AP_MOTORS_MOT_6,  -30, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  6)
 };
+#endif  // AP_MOTORS_FRAME_HEXA_ENABLED
 
+#if AP_MOTORS_FRAME_OCTA_ENABLED
```

Comment:
Sorry, you're mixing the streams here!

We try not to rely on stuff from the "other side" here.  ie. you should be able to compile support for the frame out in ArduPilot but havesim support for it - and support it in AP nbut not in SIM.

That measn creating `AP_SIM_MOTORS_FRAME_OCTAQUAD_ENABLED` etc